src/I18n/Number.php

```diff
@@ -239,28 +253,53 @@ public static function currency($value, $currency = null, array $options = [])
     /**
      * Getter/setter for default currency
      *
-     * @param string|bool|null $currency Default currency string to be used by currency()
+     * @param string|false|null $currency Default currency string to be used by currency()
      * if $currency argument is not provided. If boolean false is passed, it will clear the
      * currently stored value
      * @return string|null Currency
      */
     public static function defaultCurrency($currency = null)
     {
-        if (!empty($currency)) {
-            return self::$_defaultCurrency = $currency;
+        if ($currency === false) {
+            return static::$_defaultCurrency = null;
         }
 
-        if ($currency === false) {
-            return self::$_defaultCurrency = null;
+        if ($currency !== null) {
+            return static::$_defaultCurrency = $currency;
         }
 
-        if (empty(self::$_defaultCurrency)) {
+        if (static::$_defaultCurrency === null) {
             $locale = ini_get('intl.default_locale') ?: static::DEFAULT_LOCALE;
             $formatter = new NumberFormatter($locale, NumberFormatter::CURRENCY);
-            self::$_defaultCurrency = $formatter->getTextAttribute(NumberFormatter::CURRENCY_CODE);
+            static::$_defaultCurrency = $formatter->getTextAttribute(NumberFormatter::CURRENCY_CODE);
+        }
+
+        return static::$_defaultCurrency;
+    }
+
+    /**
+     * Getter/setter for default currency format
+     *
+     * @param string|false|null $currencyFormat Default currency format to be used by currency()
```

Comment:
This is a complex type that we can't express with a typehint. Would we be able to restrict this to `string|null`? Where null clears the current value and string values are stored? By providing `FORMAT_CURRENCY` it is straightforward to set a useful format option and `defaultCurrencyFormat(Number::FORMAT_CURRENCY)` is simpler to understand than `defaultCurrencyFormat(null)`

Reply:
Actually, why has the `defaultCurrency` function (which I modelled this on) not been deprecated and replaced with a separate getter/setter pair? That's probably the better solution here?

Reply:
Likely an oversight as we had a ton of methods to change. We could add the get/set pair for the new feature and add a deprecation in a separate pull request.

Reply:
You can probably now rebase on top of 3.next.